src/platform/freertos/memory.cpp

```
@@ -57,6 +59,31 @@ static void ensure_pool_init() {
 
+void release_message_impl(someip::Message* msg) {
+    if (!msg) {
+        return;
+    }
+
+    auto* raw = reinterpret_cast<char*>(msg);
+    auto* pool_start = static_cast<char*>(pool_buffer);
+    if (raw < pool_start || raw >= pool_start + POOL_SIZE * sizeof(someip::Message)) {
+        return;
+    }
+    size_t const byte_offset = static_cast<size_t>(raw - pool_start);
```

Cppcheck (2.20.0)

[style] The function 'get_message_id' is never used.

(unusedFunction)